### PR TITLE
Issue/14/quantize

### PR DESCRIPTION
**Pull request by aimalz**
I implemented the quantize function according to issue #14 but have some remaining issues before a merge is advised.

I couldn't get the demo to work with the class file in a different folder because I don't really know how to make packages; for now, I left the class file in the same directory as the demo.  

I changed the name of the demo notebook using 'git mv', but for some reason GitHub thinks I deleted the old notebook and made a new one, which is not what I wanted.  

The updated demo notebook works and I want to use it as a test, but I'm not sure if I have to update the demo with Travis or if Travis will know that the demo notebook has changed.  

I should probably have more questions. . .  Hopefully I labeled this correctly, too.

### CODE REVIEW

**Comment by drphilmarshall** [comment]
Good! Sounds like you should make a start on a qp folder with the required
contents, so that we can import qp.

Renaming files is the same as unix mv, and git tracks the history correctly
so no need to worry. You should edit the Travis file so that the correct
notebook is run though!

I'll review this PR tomorrow, let me know when Travis has succeeded!

**Comment by drphilmarshall** [comment]
OK, somehow conflicts had been introduced between `demo.ipynb` on master and `demo.ipynb` on this `issue/14/quantize` branch. This happens when the same lines of the file are edited in both branches - which means for us, since we are protecting the master branch, another pull request caused the  master branch `demo` to update. In general this is a _good thing_, because the master branch code _always works_. When working in a development branch, you have to be humble, and accept that the master branch code is always better :-) So, the solution is always to merge in code from master and fix the conflicts, locally. 

Here is what I did:

```
git checkout master
git pull origin master
git checkout issue/14/quantize
git merge master
atom .
... ... edit files ... ...
git status
git add demo.ipynb
git commit -am "Merged master into this branch, fixed conflicts, added autoreload"
git push origin issue/14/quantize
```

The conflicts in the notebook were not hard to fix - I just chose all the ones marked `HEAD`, because they corresponded to the code that you wrote most recently. 

I am going to make one more edit, now that the checks are passing: let's remove all that commented out code so that it does not confuse the reader (ie, us). It belongs in some sort of archive, where we can look it up and copy bits out of, but it's a distraction having it in the notebook. We should be keeping our notebook clean, so that we can _think_.

File: pdf.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        return
+
+    def integrate(self, limits):#CDF
+
+        return
+
+    def quantize(self, num_points):#inverse CDF (PPF)
+
+        quanta = 1./num_points
```
Comment (drphilmarshall) [inline pdf.py]: This needs a +1 in the denominator - I'll patch it.

**Review by drphilmarshall — COMMENTED** [review]
So far so good - I'll submit some suggestions in a PR aimed at this branch. In general, _all_ Class methods need docstrings, I'll send an example.  The `__init__` method docstring is where you put the overall Class explanation. You should Google "numpy docstring" and browse some examples.

Also, I think you are going to want to specify which quantiles to make via their separation, as an alternative. I'll make a suggestion in my patch.

**Comment by aimalz** [comment]
Since this was resolved on #17 , I'm going to close this PR.

**Comment by aimalz** [comment]
Whoops, I didn't catch that demo.ipynb had not been updated, so of course it passes the tests and nothing from this branch is making it to master.  I'm re-opening this until I figure out what I'm doing.

**Comment by aimalz** [comment]
Okay, now things seem to be correct.  I accidentally deleted a comma when manually merging, and it broke everything.  Now that it's been replaced, the tests pass with an updated demo.ipynb, and pdf.py shows your updates.  I'm leaving the branch un-deleted in case you spot something I missed.

**Comment by drphilmarshall** [comment]
The only thing I see that is wrong is that all the commented out PDF code is still in the notebook - I'll fix that.

**Pull request merged**